gdax/public_client.py

```diff
@@ -8,69 +8,165 @@
 
 
 class PublicClient(object):
-    def __init__(self, api_url="https://api.gdax.com", product_id="BTC-USD"):
-        self.url = api_url.rstrip("/")
+    """GDAX public client API.
+
+    All requests default to the `product_id` specified at object creation if not otherwise specified.
+
+    Attributes:
+        url (str): API endpoint URL
+        product_id (str): Default product used for requests
+
+    """
+
+    def __init__(self, api_url='https://api.gdax.com', product_id='BTC-USD'):
```

Comment:
Doesn't this break your principle for required parameters? I agree these are sane defaults, but if these make sense then the principle may not be completely applicable.